File: src/2022/Day1/solution.ts
import fs from "fs/promises";

async function main(file: string) {
  const data = await fs.readFile(file, { encoding: "utf-8" });
  const input = data.split("\r\n\r\n")
                .map((elfCals) => elfCals.split("\r\n").map(Number));

  const elfCalTotals = input.map((elfCals) => sumOfCalories(elfCals));
  const top3ElfCals = getTop3MostCalories(elfCalTotals);

  return {
    PartA: top3ElfCals[0],
    PartB: sumOfCalories(top3ElfCals),
  };
}

function getTop3MostCalories(totals: Array<number>): Array<number> {
  return totals.sort((a, b) => b - a).slice(0, 3);
}

function sumOfCalories(calsArr: Array<number>): number {
  return calsArr.reduce((total, cals) => total + cals);
}

main("input.txt").then((res) => console.log(res));

File: src/2022/Day1/solution.py
import heapq

with open('input.txt') as f:
    data = f.read()

data = data.replace('\r\n', '\n') # Just in case I run this in Windows
data = data.split("\n\n")

elves = []
for elf in data:
    elfCals = [int(cals) for cals in elf.split("\n")]
    elves.append(sum(elfCals))

top3Elves = heapq.nlargest(3, elves)

print("Part A: ", max(elves))
print("Part B: ", sum(top3Elves))